projects/packages/forms/src/contact-form/class-util.php

```diff
@@ -396,35 +396,45 @@ public static function grunion_contact_form_apply_block_attribute( $content, $ne
 		if ( false === stripos( $content, 'wp:jetpack/contact-form' ) ) {
 			return $content;
 		}
-		return preg_replace_callback(
-			'/<!--\s+(?P<closer>\/)?wp:jetpack\/?contact-form\s+(?P<attrs>{(?:(?:[^}]+|}+(?=})|(?!}\s+\/?-->).)*+)?}\s+)?(?P<void>\/)?-->/s',
-			function ( $match ) use ( $new_attr ) {
-				// Ignore block closers.
-				if ( ! empty( $match['closer'] ) ) {
-					return $match[0];
-				}
-				// If block doesn't have attributes, add our own.
-				if ( empty( $match['attrs'] ) ) {
-					return str_replace(
-						'wp:jetpack/contact-form ',
-						'wp:jetpack/contact-form ' . wp_json_encode( $new_attr ) . ' ',
-						$match[0]
-					);
-				}
-				$attrs = json_decode( rtrim( $match['attrs'], ' ' ), true );
-				// Ensure $attrs is an array before merging (json_decode can return null on invalid JSON).
-				if ( ! is_array( $attrs ) ) {
-					$attrs = array();
-				}
-				$attrs = array_merge( $attrs, $new_attr );
-				return str_replace(
-					$match['attrs'],
-					wp_json_encode( $attrs ) . ' ',
-					$match[0]
+
+		// Parse blocks using WordPress core function.
+		$blocks = parse_blocks( $content );
+
+		// Recursively modify contact form blocks.
+		$modified_blocks = self::modify_contact_form_blocks_recursive( $blocks, $new_attr );
```

Comment:
Why do we need to do this recursively? 

As far as I know, user can't place one form inside another. And if they do this by copying and pasting we should probably throw them an error that tells them that they need to fix it and. Was this the case before?

I wasn't able to find a test that checks this feature.

Reply:
After looking at the code, I think I understand the recursion here better and why it is needed. 

We want to look though the whole block free to find any contact form blocks. Since the first block of content might not be a form at all we have to look inside the innerblocks as well.